app/src/main/java/com/example/francescovalela/androidtutorialwinter2018/Calculator.kt

```diff
@@ -1,30 +1,94 @@
 package com.example.francescovalela.androidtutorialwinter2018
 
-import java.util.*
+import java.util.Stack
 
 class Calculator {
 
     val formula: Stack<FormulaItem> = Stack()
 
     fun pushOperand(operand: Double): Unit {
-        // TODO
+        formula.push(FormulaItem(operand = operand))
     }
 
     fun pushOperator(operator: Char): Unit {
-        // TODO
+        if (formula.isEmpty())
```

Comment:
do you really need this?
Wouldn't
```kotlin
if (!formula.peek().isOperand) {
  formula.pop()
  formula.push(FormulaItem(operator = operator))
```
Be enough?

Reply:
Absolutely. New PR incoming.